### ISSUE TITLE
Stick-dim work division collapses to one core when the dim size isn't a multiple of the division

### ISSUE THREAD

**Issue by yoheiueda** [torch-spyre/torch-spyre#3339]
### What happened?

 pointwise op whose stick (last) dimension spans multiple sticks silently miscompiles when work division divides that stick dimension across multiple cores. Every element past the first stick is corrupted.

`align_tensors` (`torch_spyre/_inductor/views.py`) re-intersects work division's committed per-dimension core split against a variable's range via `gcd`. For a stick dimension, work division commits its core count against the **stick count** (`adjust_it_space_for_sticks`), but `align_tensors` used the **element count** as the `gcd` basis. When the element count is coprime with the committed split — e.g. 67 elements over a 2-stick / 2-core split, `gcd(2, 67) == 1` — the split collapses to a single core, so a single core now owns a fold that spans both sticks.

That collapsed layout only produces wrong output when both operands are **LX-resident** (`LX_PLANNING` on, which is the default). The LX branch of `_start_addr_data` gives every core the same buffer base address, so the single core's second-stick loads land in the wrong region. With `LX_PLANNING=0` the operands live in HBM, whose per-core addressing resolves the second stick correctly even though the split still collapses — so LX planning is a necessary condition for the miscompile, not the cause. A common pointwise producer (`neg`, `mul`, any op eligible for LX reuse) puts its result in LX under the default config, so the default path hits it readily.

Fires for any stick-dim extent `N > 64` that is coprime with the committed core split. Multiples of `64 x split` (e.g. 128, 192, 256) are clean. In multi-dimensional shapes the trigger requires the outer dims to be small enough that work division divides the stick dim rather than the outer dims.

Confirmed 1-D patterns (last dim = stick dim), where `sticks = ceil(N / 64)` and `split` is the core count work division commits to the stick dim:

| N   | sticks | split | gcd(split, N) | result       |
| --- | ------ | ----- | ------------- | ------------ |
| 64  | 1      | 1     | 1             | clean        |
| 65  | 2      | 2     | 1             | **miscompile** |
| 66  | 2      | 2     | 2             | clean        |
| 67  | 2      | 2     | 1             | **miscompile** |
| 95  | 2      | 2     | 1             | **miscompile** |
| 97  | 2      | 2     | 1             | **miscompile** |
| 127 | 2      | 2     | 1             | **miscompile** |
| 128 | 2      | 2     | 2             | clean        |
| 130 | 3      | 3     | 1             | **miscompile** |
| 160 | 3      | 3     | 1             | **miscompile** |
| 191 | 3      | 3     | 1             | **miscompile** |
| 192 | 3      | 3     | 3             | clean        |
| 193 | 4      | 4     | 1             | **miscompile** |
| 255 | 4      | 4     | 1             | **miscompile** |
| 256 | 4      | 4     | 4             | clean        |
| 257 | 5      | 5     | 1             | **miscompile** |

The miscompile fires exactly when `gcd(split, N) < split` (i.e. `N` coprime with `split`), which collapses the committed multi-core split to `gcd(split, N)` cores.

Confirmed multi-dimensional patterns (last dim = stick dim):

| shape           | stick dim divided across cores? | result       |
| --------------- | ------------------------------- | ------------ |
| `(3, 67)`       | yes                             | **miscompile** |
| `(4, 193)`      | yes                             | **miscompile** |
| `(2, 3, 67)`    | yes                             | **miscompile** |
| `(3, 5, 127)`   | yes                             | **miscompile** |
| `(2, 3, 2, 127)`| yes                             | **miscompile** |
| `(2, 129)`      | no (outer dim absorbs split)    | clean        |
| `(2, 2, 129)`   | no (outer dim absorbs split)    | clean        |
| `(2, 3, 4, 67)` | no (outer dim absorbs split)    | clean        |

### What did you expect to happen?

The compiled result to match CPU. The split that work division already certified against the stick count (2 cores, one stick each) should survive iteration-space realignment, so each stick lands on its own core.

### How can we reproduce it?

Default configuration (multi-core, `sencores=32`, `LX_PLANNING` on). The repro below uses `torch.neg` on both operands so their results are LX-resident; a plain `x + y` of two graph inputs keeps the operands in HBM and does not fire.

Two settings independently mask it:

- `SENCORES=1` — single core, so there is no split to collapse (removes the collapse).
- `LX_PLANNING=0` — operands live in HBM, whose per-core addressing resolves the second stick correctly (removes the LX single-base addressing). The split still collapses under this setting.

```python
import torch

def fn(x, y):
    return torch.neg(x) + torch.neg(y)

# 67 elements = 2 sticks (64 + 3), coprime with the 2-core split.
# Integer values <= 1024 are exact in device fp16, so this is a real mismatch.
n = 67
x = (torch.arange(n) % 251).to(torch.float16)
y = ((torch.arange(n) + 100) % 251).to(torch.float16)

got = torch.compile(fn, dynamic=False)(x.to("spyre"), y.to("spyre")).cpu()
ref = fn(x, y)
torch.testing.assert_close(got, ref)   # fails: elements 64..66 corrupted
```

### Any environmental details we need to know?

- torch-spyre: `origin/main` at `0883cb6`
- PyTorch: `2.11.0`

### Anything else we need to know?

Root cause is a pure basis mismatch: `work_division.py` already uses the stick count as the split basis for stick vars (`enumerate_work_division_candidates`), then `align_tensors` re-intersects the same split against the element count — a basis its own upstream never uses for a stick var. Fix: feed the stick-count basis (`ceil(elems / elems_per_stick)`) into the `gcd` for the stick var's segment; non-stick vars keep the element basis.
